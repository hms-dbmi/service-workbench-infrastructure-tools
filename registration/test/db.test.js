const { getUserParams } = require('../db');

process.env.USER_TABLE = 'some-table';
process.env.USER_REASON = 'testing fun';
process.env.USER_POOL = 'some-user-pool-id';
process.env.IDP_NAME = 'Authorizer';

describe('getUserParams', () => {
  const testData = {
    uid: '12345',
    firstName: 'Test',
    lastName: 'Tester',
    email: 'test@email.com',
    createdAt: '2022-08-17T16:28:33.316Z'
  };

  it('should not throw an error', () => {
    expect(() => getUserParams(testData)).not.toThrowError();
  });

  it('should return correctly formatted object for dynamodb insertion', () => {
    const params = getUserParams(testData);
    expect(params).toMatchSnapshot();
  });
});