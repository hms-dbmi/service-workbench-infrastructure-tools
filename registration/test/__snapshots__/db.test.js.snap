// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUserParams should return correctly formatted object for dynamodb insertion 1`] = `
Object {
  "Item": Object {
    "applyReason": Object {
      "S": "testing fun",
    },
    "authenticationProviderId": Object {
      "S": "some-user-pool-id",
    },
    "createdAt": Object {
      "S": "2022-08-17T16:28:33.316Z",
    },
    "createdBy": Object {
      "S": "_system_",
    },
    "email": Object {
      "S": "test@email.com",
    },
    "encryptedCreds": Object {
      "S": "N/A",
    },
    "firstName": Object {
      "S": "Test",
    },
    "identityProviderName": Object {
      "S": "Authorizer",
    },
    "isAdmin": Object {
      "BOOL": false,
    },
    "isExternalUser": Object {
      "BOOL": true,
    },
    "isNativePoolUser": Object {
      "BOOL": false,
    },
    "isSamlAuthenticatedUser": Object {
      "BOOL": true,
    },
    "lastName": Object {
      "S": "Tester",
    },
    "ns": Object {
      "S": "Authorizer||||some-user-pool-id",
    },
    "projectId": Object {
      "L": Array [],
    },
    "rev": Object {
      "S": "1",
    },
    "status": Object {
      "S": "pending",
    },
    "uid": Object {
      "S": "12345",
    },
    "userRole": Object {
      "S": "",
    },
    "username": Object {
      "S": "test@email.com",
    },
  },
  "TableName": "some-table",
}
`;
